### DISCUSSION TITLE
Word count

### DISCUSSION THREAD

**Discussion by DuckDuckWhale** [helix-editor/helix#10211] [Ideas]
In Vim, `g<C-g>` would show `Col 8 of 274; Line 3 of 3; Word 6 of 52; Byte 35 of 302` in the bottom left corner. It would be nice if helix also has some functionality to count words.

**Comment by swork1** [comment]
> In Vim, `g<C-g>` would show `Col 8 of 274; Line 3 of 3; Word 6 of 52; Byte 35 of 302` in the bottom left corner. It would be nice if helix also has some functionality to count words.

What about implementing this as an [`editor.statusline`](https://docs.helix-editor.com/master/configuration.html#editorstatusline-section) element. Such as `word-count`, allowing a user to add this to their `config.toml` file?

**Comment by archseer** [comment]
We already have line/col information, but word count isn't as cheap to compute during editing since we'd need to rescan each time a change is made.

**Comment by DuckDuckWhale** [comment]
Would it be possible to track the delta in word count every time an edit is made? Something like if the edit introduces more whitespace sections, increase the word count by that number, and decrease it if it removes whitespace sections? I realize that it would be non-trivial to implement and therefore might not be the best way, but it might be theoretically possible without being too expensive to compute.

**Comment by codeswish** [comment]
> We already have line/col information, but word count isn't as cheap to compute during editing since we'd need to rescan each time a change is made.

The ```g<C-g>``` command only provides the information based on the cursor's position when it is invoked. It is not continuously updated while editing. Wouldn't this mean a scan would only need to occur when the command is given?

**Comment by thairanaru** [comment]
I also agree that having builtin feature would be nice. 
For a workaround, I have a marco shortcut which pipes a selection into a script that uses wc -w and notify-send to show a notification listing the word count. I hope this helps for people finding this thread.

**Reply by yashvesikar** [reply]
Thanks for the idea! I ended up binding it through a custom command in my `config.toml`: 
```
[keys.normal.space]

# Custom command for word count
A-w = "@<A-|>tee /tmp/helix-wc.file<ret>:sh cat /tmp/test.file | wc -w<ret>"
 ```

**[source](https://www.reddit.com/r/HelixEditor/comments/1h1w8dh/comment/lzmtd7o)**

**Reply by BlameTroi** [reply]
> Thanks for the idea! I ended up binding it through a custom command in my `config.toml`:
> 
> ```
> [keys.normal.space]
> 
> # Custom command for word count
> A-w = "@<A-|>tee /tmp/helix-wc.file<ret>:sh cat /tmp/test.file | wc -w<ret>"
> ```

Minor typo here, correcting in case someone else just yanks the code without looking at it too closely first.

```
[keys.normal.space]
# Custom command for word count of selection.
A-w = "@<A-|>tee /tmp/helix-wc.file<ret>:sh cat /tmp/helix-wc.file | wc -w<ret>"
```
